### ISSUE TITLE
Error: Type … must define one or more fields.

### ISSUE THREAD

**Issue by langpavel** [cjoudrey/graphql-schema-linter#151]
Hi. I have file which looks like this:
```graphql
# You should write `extend type Mutation` in relevant places
"""
Main Mutation Type
"""
type Mutation
```

This is really correct GraphQL SDL. Use case is obvious.

But this file is source this error:
```
$ graphql-schema-linter "schema/**/*.gql"
It looks like you may have hit a bug in graphql-schema-linter.

It would be super helpful if you could report this here: https://github.com/cjoudrey/graphql-schema-linter/issues/new

Error: Type Mutation must define one or more fields.
```

even if I have file which `extend type Mutation { … }` somewhere else.

Reason why I want this file is to be able to *comment on types* and to *point other developers to think differently — more modular*

**Comment by cjoudrey** [comment]
Thanks for opening this issue @langpavel and sorry for the radio silence.

That's really interesting. 🤔This error seems to be coming [from GraphQL.js](https://github.com/graphql/graphql-js/blob/f4dee2803c22f7e5fde913b9743cc6de345bb871/src/type/validate.js#L275-L281).

Out of curiosity, what GraphQL library do you use to serve this schema?

**Comment by langpavel** [comment]
Hi, I'm using `graphql` — reference implementation to preprocess multiple GraphQL files and then using output in Apollo Server.

I'm merging all files together just by concatenation of `definitions` arrays together,
then I do some magic before using schema in Apollo — like merging extensions on AST level
without using `extendSchema` API.

That may be source of this error, because I never have empty type definition before instantiating schema.

On the other way, when I declare empty type I can see exactly same error coming from `assertValidSchema`

## TL;DR: Summary
 - It is valid syntax to declare type without fields, AST will be generated
 - It is not valid to instantiate schema which contains empty types (even if AST contains `extend type`)
 - `graphql` does not contain API to extend schema before instantiating

**Comment by aldeed** [comment]
I'm seeing the same thing with `union Blah` where this union is extended in other files. Apollo Server does not complain about this and it works fine using the same version of the `graphql` package.

This package uses `validateSDL`, which is marked as an internal function in the `graphql` package. It seems that the behavior is difference using `validateSDL` directly rather than `validate`.
